### PR TITLE
[SYCL-MLIR] XFAIL other lits failed on other platforms

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Some tests were missing from the xfail list in https://github.com/intel/llvm-test-suite/pull/1536.

Signed-off-by: Tsang, Whitney <whitney.tsang@intel.com>

### CODE REVIEW

**Comment by victor-eds** [comment]
Are these tests passing on CPU but failing on GPU?

**Comment by victor-eds** [comment]
> Are these tests passing on CPU but failing on GPU?

Okey, I see now. If these failures depend on the platform, we might need different XFail lists for different platforms. If we merge this change, the tests won't pass for `OCL x64`, right?

**Comment by whitneywhtsang** [comment]
> > Are these tests passing on CPU but failing on GPU?
> 
> Okey, I see now. If these failures depend on the platform, we might need different XFail lists for different platforms. If we merge this change, the tests won't pass for `OCL x64`, right?

I ran this updated list on my machine, and is still passing, likely all because they are unsupported on cpu.
I checked some of them, and they are indeed marked with `UNSUPPORTED: cpu`, or don't have run line for CPU.

**Comment by whitneywhtsang** [comment]
Updated the list to only add the tests that fail or unsupported on all platforms our buildbots run.
The buildbot would still be failing now, as there are still tests that pass on some platforms and fail on the others.

File: SYCL/xfail_tests.txt [hunk 1]
```diff
@@ -262,8 +273,8 @@ SubGroup/shuffle.cpp
 SubGroup/sub_group_as.cpp
 SubGroup/sub_group_as_vec.cpp
 SubGroup/sub_groups_sycl2020.cpp
-SubGroup/vote.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:265]: Pass on all 3 buildbots, can remove from the xfail list.

File: SYCL/xfail_tests.txt [hunk 2]
```diff
@@ -127,6 +127,10 @@ Complex/sycl_complex_math_test.cpp
 Complex/sycl_complex_operator_test.cpp
 Complex/sycl_complex_pow_test.cpp
 Complex/sycl_complex_stream_test.cpp
+DeviceGlobal/device_global_arrow.cpp
+DeviceGlobal/device_global_device_only.cpp
+DeviceGlobal/device_global_operator_passthrough.cpp
+DeviceGlobal/device_global_subscript.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:133]: Unsupported on x64. Failed on GEN9.

File: SYCL/xfail_tests.txt [hunk 3]
```diff
@@ -150,6 +154,8 @@ DiscardEvents/discard_events_check_images.cpp
 DiscardEvents/discard_events_using_assert.cpp
 DotProduct/dot_product_int_test.cpp
 DotProduct/dot_product_vec_test.cpp
+ESIMD/dword_atomic_cmpxchg.cpp
+ESIMD/dword_atomic_smoke.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:158]: Unsupported on x64. Failed on GEN9.

File: SYCL/xfail_tests.txt [hunk 4]
```diff
@@ -178,6 +184,8 @@ KernelParams/array-kernel-param-run.cpp
 KernelParams/union_kernel_param.cpp
 OptionalKernelFeatures/is_compatible.cpp
 Plugin/enqueue-arg-order-image.cpp
+Plugin/interop-level-zero-buffer-multi-dim.cpp
+Plugin/level_zero_track_indirect_access_memory.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:188]: Unsupported on OCL GEN9 and x64. Failed on L0 GEN9.

File: SYCL/xfail_tests.txt [hunk 5]
```diff
@@ -216,8 +224,11 @@ Reduction/reduction_usm_dw.cpp
 Reduction/regression_after_pr_6343.cpp
 Regression/atomic_load.cpp
 Regression/group.cpp
+Regression/half_operators.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:227]: Unsupported on x64. Failed on GEN9.

File: SYCL/xfail_tests.txt [hunk 6]
```diff
@@ -216,8 +224,11 @@ Reduction/reduction_usm_dw.cpp
...
 Regression/implicit_atomic_conversion.cpp
 Regression/implicit_kernel_bundle_image_filtering.cpp
+Regression/reduction_resource_leak_dw.cpp
+Regression/reduction_resource_leak_usm.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:231]: Unsupported on OCL GEN9 and x64. Failed on L0 GEN9.

File: SYCL/xfail_tests.txt [hunk 7]
```diff
@@ -262,8 +273,8 @@ SubGroup/shuffle.cpp
 SubGroup/sub_group_as.cpp
 SubGroup/sub_group_as_vec.cpp
 SubGroup/sub_groups_sycl2020.cpp
-SubGroup/vote.cpp
 SubGroupMask/Basic.cpp
+SubGroupMask/GroupSize.cpp
```
Comment (whitneywhtsang) [inline SYCL/xfail_tests.txt:277]: Unsupported on x64. Failed on GEN9.

**Comment by victor-eds** [comment]
Oh, great! We should merge in that case. Thanks!

**Review by victor-eds — APPROVED** [review]

**Pull request merged**